modules/os-datagrid-online-services-configuration/src/main/bash/jdg-online-configuration.sh

```diff
@@ -53,6 +55,10 @@ done
 check_if_profile_exists
 check_if_cli_exists
 
+if [ ${PROFILE} = "SHARED-MEMORY-SERVICE" ]; then
+    echo "set num_owners=${NUMBER_OF_OWNERS}" > $CLI_PROPERTIES_FILE
```

Comment:
I've stumbled upon something very similar recently. I was struggling how to pass parameters from shell script into Wildfly CLI. 

I did something slightly different here [[link to commit](https://github.com/slaskawi/jboss-datagrid-services-image/commit/36200486eec829df11d459dc3cde3ef18ff0a6d2)] - I changed the way the CLI script separates arguments (previously it was `key=val`) to `key [space] val` and I forward to the Infinispan CLI all not parsed arguments. This way you can invoke the CLI like this: `$CLI --file="${PROFILES_DIRECTORY}/${PROFILE}.cli" -Dmykey=myval -Dmykey2=myval`. 

In general I prefer your way to be honest. But passing configuration parameters using environmental variables will be very frequently used in our script, so I think using `-D` option might work slightly better. But that's not a very strong opinion TBH. If you prefer to leave it like this I can revert my changes. 

WDYT?

Reply:
@slaskawi TBH I don't really mind either way. I'm happy to change to your approach as it is more explicit than my approach and does not need the reader to have knowledge of `.jbossclirc` and the jboss cli.

Reply:
I'll update the script next week once DR3 is ready to be pulled

Reply:
It seems I was wrong here. For some reason I can not get it working with `-D` option (I have no idea why I though it works). So let me take back all my words! - Let's use `$CLI_PROPERTIES_FILE` as you suggested.

Reply:
ooh man... the more I'm digging for this the crazier solutions I find: https://access.redhat.com/solutions/321513. The most scary thing is that they work!

Reply:
no, no it's too scary... I'm implementing this your way ;)